tuner/tuner/common.py

```diff
@@ -266,3 +268,41 @@ class MLIRTransformation:
     template: list[str]
     modified: str
     embeddable: str
+
+
+def combine_tuning_specs(
+    tuner_ctx: TunerContext, td_specs: list[ir.Module]
+) -> ir.Module:
+    with tuner_ctx.mlir_ctx as ctx, ir.Location.unknown():
+        top_module = ir.Module.create()
+        top_module.operation.attributes[
+            "transform.with_named_sequence"
+        ] = ir.UnitAttr.get()
+
+        for td_spec in td_specs:
+            top_module.body.append(td_spec.operation.clone())
+        return top_module
+
+
+def link_tuning_specs(tuner_ctx: TunerContext, td_specs: list[ir.Module]) -> ir.Module:
+    module = combine_tuning_specs(tuner_ctx, td_specs)
+    iree_opt = ireec.binaries.find_tool("iree-opt")
+
+    result = subprocess.run(
+        [
+            iree_opt,
+            "--split-input-file",
+            "--verify-diagnostics",
+            "--iree-codegen-link-tuning-specs",
+            "-",
+        ],
+        input=str(module),  # Provide MLIR text directly via stdin.
```

Comment:
I think this may be very annoying to reproduce if it ever fails. Why not serialize it to disk and put under a temp directory?